ta/pkcs11/src/object.c

```diff
@@ -291,8 +291,10 @@ enum pkcs11_rc create_object(void *sess, struct obj_attrs *head,
 	handle_put(get_object_handle_db(session), obj_handle);
 	if (get_bool(head, PKCS11_CKA_TOKEN))
 		cleanup_persistent_object(obj, session->token);
-	else
+	else {
+		LIST_REMOVE(obj, link);
```

Comment:
Thanks for the finding. It does the job but I think we could end up calling `LIST_REMOVE(obj, link)` only from `cleanup_volatile_obj_ref()`. It would replace this added call and those at lines 122 and 143. What do you think?

Reply:
cleanup_volatile_obj_ref() is also called inside cleanup_persistent_object().  LIST_REMOVE() will be called multiple times for token obj especially in the destroy_object(). I do not think it is a good behavior although it does not create error for now. Meanwhile, obj is explicitly inserted to list inside create_object(), it is much clearer to read if it is also removed inside the same function.

Reply:
Yeah, it is a good idea. Let me update the patch.